crates/router/src/db/file.rs

```diff
@@ -0,0 +1,92 @@
+use error_stack::IntoReport;
+
+use super::{MockDb, Store};
+use crate::{
+    connection,
+    core::errors::{self, CustomResult},
+    types::storage,
+};
+
+#[async_trait::async_trait]
+pub trait FileInterface {
```

Comment:
I would suggest changing this to `FileMetadataInterface` and in all the other places too. This would avoid confusion as to whether the file or file metadata is being inserted in the db. Please rename all the related methods too.